test/projectDataProvider.test.ts

```diff
@@ -21,96 +22,119 @@ suite('projectDataProvider Modules', () => {
-        ProjectDataProvider.effectivef8Pom('path/pom.xml', (cb) => {
-            expect(cb).equals('path/target/pom.xml');
-        });
+        let effectivef8PomPR = await ProjectDataProvider.effectivef8Pom('path/pom.xml');
+        expect(effectivef8PomPR).equals('path/target/pom.xml');
         expect(stubExec).calledOnce;
     });
 
     test('effectivef8Package should return error', () => {
-        let stubGetDependencyVersion = sandbox.stub(ProjectDataProvider, 'getDependencyVersion').yields(false);
-        ProjectDataProvider.effectivef8Package('path/package.json', (cb) => {
-            expect(cb).equals(false);
-        });
+        let workspaceFolder = vscode.workspace.workspaceFolders[0];
+        let stubGetDependencyVersion = sandbox.stub(ProjectDataProvider, 'getDependencyVersion').rejects(false);
+        ProjectDataProvider.effectivef8Package(workspaceFolder);
         expect(stubGetDependencyVersion).calledOnce;
     });
 
-    test('effectivef8Package should return success', () => {
-        let stubGetDependencyVersion = sandbox.stub(ProjectDataProvider, 'getDependencyVersion').yields(true);
-        let stubFormPackagedependency = sandbox.stub(ProjectDataProvider, 'formPackagedependencyNpmList').resolves('sample');
-        ProjectDataProvider.effectivef8Package('path/package.json', (cb) => {
-            expect(cb).equals('sample');
-        });
+    test('effectivef8Package should return success', async () => {
+        let workspaceFolder = vscode.workspace.workspaceFolders[0];
+        let stubGetDependencyVersion = sandbox.stub(ProjectDataProvider, 'getDependencyVersion').resolves(true);
+        let stubFormPackagedependencyNpmList = sandbox.stub(ProjectDataProvider, 'formPackagedependencyNpmList').resolves('sample');
+        await ProjectDataProvider.effectivef8Package(workspaceFolder);
         expect(stubGetDependencyVersion).calledOnce;
-        expect(stubFormPackagedependency).calledOnce;
+        expect(stubFormPackagedependencyNpmList).calledOnce;
     });
 
-    test('effectivef8Package should return error if formPackagedependencyNpmList fails', () => {
-        let stubGetDependencyVersion = sandbox.stub(ProjectDataProvider, 'getDependencyVersion').yields(true);
-        let stubFormPackagedependency = sandbox.stub(ProjectDataProvider, 'formPackagedependencyNpmList').rejects('err');
-        ProjectDataProvider.effectivef8Package('path/package.json', (cb) => {
-            expect(cb).equals(false);
-        });
-        expect(stubGetDependencyVersion).calledOnce;
-        expect(stubFormPackagedependency).calledOnce;
-    });
-
-    test('formPackagedependency should return error', () => {
+    test('formPackagedependencyNpmList should return error', () => {
+        let workspaceFolder = vscode.workspace.workspaceFolders[0];
+        let vscodeRootpath = workspaceFolder.uri.fsPath;
+        if(process && process.platform && process.platform.toLowerCase() === 'win32'){
```

Comment:
can you write a separate test for different platforms like win32, darwin and linux? and can you stub the path?